### PR TITLE
Move to github actions

### LINKED ISSUE #787: Travis CI free usage ends Dec 3; mozilla repos should switch to other CI platforms
We're opening this issue because your project has used Travis CI within the last 6 months. If you have already migrated off it, you can close and ignore this issue.

Travis CI is ending free builds on public repositories. travis-ci.com [stopped providing](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing)them in early November, and travis-ci.org [will stop](https://mailchi.mp/3d439eeb1098/travis-ciorg-is-moving-to-travis-cicom) after December 31, 2020. To avoid disruptions to your workflows, you must migrate to another CI service.

For production use cases, we recommend switching to [CircleCI](https://circleci.com/). This service is already widely used within Mozilla. There is a guide to migrating from Travis CI to CircleCI available [here](https://circleci.com/docs/2.0/migrating-from-travis/).

For non production use cases, we recommend either CircleCI or [Github Actions](https://github.com/features/actions). There is a guide to migrating from Travis CI to Github Actions available [here](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions). Github Actions usage within Mozilla is new, and you will have to work with our github administrators to enable specific actions following [this process](https://wiki.mozilla.org/GitHub#How_do_I_hook_up_a_new_GitHub_Actions_or_3rd_party_application_to_a_repository_in_the_mozilla_org.3F).

If you have any questions, reach out in `#github-admin:mozilla.org` on [matrix](https://chat.mozilla.org/#/room/#github-admin:mozilla.org).

### PR DESCRIPTION
**Pull request by autrilla**
Basically reopening #576 now that Travis CI is deprecated (#787). It doesn't handle releases right now, but if it at least means we have working tests for PRs and so on, I think that's a net win.

Fixes #787.

### CODE REVIEW

**Comment by autrilla** [comment]
This works on my fork: https://github.com/autrilla/sops/runs/1622187193

File: .github/workflows/cli.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  build:
+    name: Build ${{ matrix.os }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [linux]
+    steps:
+      - name: Install dependencies
+        run: sudo apt-get update && sudo apt-get install git -y
+      - name: Set up Go 1.13
```
Comment (ajvb) [inline .github/workflows/cli.yml:19]: Should we pull the go version from https://github.com/mozilla/sops/blob/master/go.mod#L3 ?

File: .github/workflows/cli.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Build
+        run: GOOS=${{ matrix.os }} go build -o sops-${{ matrix.os }}-${{ github.sha }} -v go.mozilla.org/sops/cmd/sops
+      - name: Upload artifact
```
Comment (ajvb) [inline .github/workflows/cli.yml:34]: Will this also do mac and windows builds?

**Review by ajvb — APPROVED** [review]
Awesome, thank you so much @autrilla!

File: .github/workflows/cli.yml (hunk 2 above)
Comment (autrilla) [reply .github/workflows/cli.yml:34]: It does now. See the artifacts in https://github.com/autrilla/sops/actions/runs/463836838

File: .github/workflows/cli.yml (hunk 1 above)
Comment (autrilla) [reply .github/workflows/cli.yml:19]: It's not trivial unless we generate the workflow yaml from that other file. I think for now it's fine to leave as is. That's how it was for the travisci config.

File: .github/workflows/cli.yml (hunk 2 above)
Comment (ajvb) [reply .github/workflows/cli.yml:34]: Awesome

File: .github/workflows/cli.yml (hunk 1 above)
Comment (ajvb) [reply .github/workflows/cli.yml:19]: Ok sounds good to me.

**Comment by autrilla** [comment]
@ajvb can you merge this manually? Right now we need TravisCI to pass, but since they ended support for open source, it never will.

**Comment by ajvb** [comment]
@autrilla Looks like our settings our not allowing the `actions-rs` action to be ran - https://github.com/mozilla/sops/actions/runs/466628788

There might be a wait to add this to an allow list, will look into it today.

**Comment by autrilla** [comment]
It should be fairly easy to replace it with rustup if getting it allowed is
not trivial.

**Comment by ajvb** [comment]
@autrilla yeah, I'd say let's go with rustup for now. The process (https://wiki.mozilla.org/GitHub#GitHub_Actions) is there, but will require some back and forth. If you want me to initiate getting `actions-rs` approved though, let me know.

**Pull request merged**